script.js

```diff
@@ -1,4 +1,160 @@
+    .trim())}.<br><br> Player 2 press Enter to roll your dice.`; // to combine the 0 and 1 index together
+};
+var diceOrder2Player1 = function () {
+  userDiceOrderPlayer1.push(diceRoll2Player1, diceRoll1Player1);
+  return `Player 1: Your number is ${(player1FinalNumb = userDiceOrderPlayer1
+    .join("")
+    .trim())}. <br><br> Player 2 press Enter to roll your dice.`; // to combine the 0 and 1 index together
+};
+
+// Rolling the dice for player 2
+var player2RollDice = function () {
+  return `Player 2: Dice 1 rolled ${diceRoll1Player2} and Dice 2 rolled ${diceRoll2Player2}. <br><br> Please input 1 or 2 to choose the order of your dice roll.`;
+};
+
+// Order of dices for player 2
+var userDiceOrderPlayer2 = [];
+var diceOrder1Player2 = function () {
+  userDiceOrderPlayer2.push(diceRoll1Player2, diceRoll2Player2);
+  return `Player 2: Your number is ${(player2FinalNumb = userDiceOrderPlayer2
+    .join("")
+    .trim())}.<br><br> Press Enter to view results.`; // to combine the 0 and 1 index together
+};
+
+var diceOrder2Player2 = function () {
+  userDiceOrderPlayer2.push(diceRoll2Player2, diceRoll1Player2);
+  return `Player 2: Your number is ${(player2FinalNumb = userDiceOrderPlayer2
+    .join("")
+    .trim())}. <br><br> Press Enter to view results.`; // to combine the 0 and 1 index together
+};
+
+// Leaderboard display
+var displayLBoard = function () {
+  if (player1Score > player2Score || player1Score == player2Score) {
+    return `LEADERBOARD <br><br> Player 1 Score: ${player1Score} <br><br> Player 2 Score: ${player2Score}<br><br> Press Enter to start another round.`;
+  } else {
+    return `LEADERBOARD <br><br> Player 2 Score: ${player2Score} <br><br> Player 1 Score: ${player1Score}<br><br> Press Enter to start another round.`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (currentMode == "start") {
+    myOutputValue = `Choose normal or reversed mode to begin 🎲`;
+  }
+  if (currentMode == "start" && input == "normal") {
+    myOutputValue = `You have chosen NORMAL mode. <br><br> Press enter to start rolling the dice 🎲`;
+    currentMode = "normal mode";
+  } else if (currentMode == "start" && input == "reversed") {
+    myOutputValue = `You have chosen REVERSED mode. <br><br> Press enter to start rolling the dice 🎲`;
+    currentMode = "normal mode";
+    reversedMode = true;
+    // normal mode will roll the dice
+  } else if (currentMode == "normal mode") {
+    diceRoll1Player1 = diceRollGenerator();
+    diceRoll2Player1 = diceRollGenerator();
+    myOutputValue = player1RollDice();
+    currentMode = "player 1 rolled";
+    // input the order
+  } else if (currentMode == "player 1 rolled" && input == 1) {
+    var player1Order = diceOrder1Player1();
+    currentMode = "player 2 start";
+    return player1Order;
+  } else if (currentMode == "player 1 rolled" && input == 2) {
+    var player1Order = diceOrder2Player1();
+    currentMode = "player 2 start";
+    return player1Order;
```

Comment:
As mentioned above these functions can potentially be combined and the input == 1, input ==2 can be incorporated into the helper function.